lib/instrumentation/modules/http2.js

```diff
@@ -127,4 +131,61 @@ module.exports = function (http2, agent) {
     }
     return result
   }
+
+  function wrapConnect (orig) {
+    return function (host) {
+      const ret = orig.apply(this, arguments)
+      shimmer.wrap(ret, 'request', orig => wrapRequest(orig, host))
+      return ret
+    }
+  }
+
+  function wrapRequest (orig, host) {
+    return function (headers) {
+      var span = agent.buildSpan()
+      var id = span && span.transaction.id
+
+      agent.logger.debug('intercepted call to http2.request %o', { id })
+
+      var req = orig.apply(this, arguments)
+      if (!span) return req
+
+      var protocol = req.agent && req.agent.protocol
+      agent.logger.debug('request details: %o', { protocol, host, id })
+
+      var method = headers[':method'] || 'GET'
+      var name = method + ' ' + host + url.parse(headers[':path']).pathname
+      span.start(name, 'ext.http2')
+      req.on('response', onResponse)
+
+      return req
+
+      function onResponse (res) {
+        agent.logger.debug('intercepted http2 client response event %o', { id })
+
+        // Inspired by:
+        // https://github.com/nodejs/node/blob/9623ce572a02632b7596452e079bba066db3a429/lib/events.js#L258-L274
+        if (res.prependListener) {
+          // Added in Node.js 6.0.0
+          req.prependListener('end', onEnd)
```

Comment:
You're not listening for the `end` event on the `res` object as we do with HTTP/1.1. Is that due to HTTP2 being different?

If that's the case we might not need the whole `onResponse` function at all if we can just move the `onEnd` stuff outside

Reply:
Yep. Actually, the `res` argument for `onResponse` is wrong...it's `headers` and `flags`. The incoming and outgoing parts of HTTP2 aren't separated like HTTP1.

Reply:
I just made some changes. You were right that the extra stuff is not needed.